x-pack/plugins/uptime/public/state/api/alert_actions.ts

```diff
@@ -0,0 +1,144 @@
+export const PAGER_DUTY_ACTION_ID: ActionTypeId = '.pagerduty';
+export const SERVER_LOG_ACTION_ID: ActionTypeId = '.server-log';
+export const INDEX_ACTION_ID: ActionTypeId = '.index';
+export const TEAMS_ACTION_ID: ActionTypeId = '.teams';
+export const SERVICE_NOW_ACTION_ID: ActionTypeId = '.servicenow';
+export const JIRA_ACTION_ID: ActionTypeId = '.jira';
+export const WEBHOOK_ACTION_ID: ActionTypeId = '.webhook';
+
+const { MONITOR_STATUS } = ACTION_GROUP_DEFINITIONS;
+
+export function populateAlertActions({ defaultActions, monitorId, monitorName }: NewAlertParams) {
+  const actions: AlertAction[] = [];
+  defaultActions.forEach((aId) => {
+    const action: AlertAction = {
+      id: aId.id,
+      actionTypeId: aId.actionTypeId,
+      group: MONITOR_STATUS.id,
+      params: {},
+    };
+    switch (aId.actionTypeId) {
+      case PAGER_DUTY_ACTION_ID:
+        action.params = getPagerDutyActionParams(monitorId);
+        break;
+      case SERVER_LOG_ACTION_ID:
+        action.params = getServerLogActionParams();
+        break;
+      case INDEX_ACTION_ID:
+        action.params = getIndexActionParams();
+        break;
+      case SERVICE_NOW_ACTION_ID:
+        action.params = getServiceNowActionParams();
+        break;
+      case JIRA_ACTION_ID:
+        action.params = getJiraActionParams();
+        break;
+      case WEBHOOK_ACTION_ID:
+        action.params = getWebhookActionParams();
+        break;
+      case SLACK_ACTION_ID:
+      case TEAMS_ACTION_ID:
+      default:
+        action.params = {
+          message: MonitorStatusTranslations.defaultActionMessage,
+        };
+    }
+
+    actions.push(action);
+  });
+
+  return actions;
+}
+
+function getIndexActionParams(): IndexActionParams {
+  return {
+    documents: [
+      {
+        monitorName: '{{state.monitorName}}',
+        monitorUrl: '{{{state.monitorUrl}}}',
+        statusMessage: '{{state.statusMessage}}',
+        latestErrorMessage: '{{{state.latestErrorMessage}}}',
+        observerLocation: '{{state.observerLocation}}',
+      },
+    ],
+  };
+}
+
+function getServerLogActionParams(): ServerLogActionParams {
+  return {
+    level: 'warn',
+    message: MonitorStatusTranslations.defaultActionMessage,
+  };
+}
+
+function getWebhookActionParams(): WebhookActionParams {
+  return {
+    body: MonitorStatusTranslations.defaultActionMessage,
+  };
+}
+
+function getPagerDutyActionParams(monitorId: string): PagerDutyActionParams {
+  return {
+    dedupKey: monitorId + MONITOR_STATUS.id,
+    eventAction: 'trigger',
+    severity: 'error',
+    summary: MonitorStatusTranslations.defaultActionMessage,
+  };
+}
+
+function getServiceNowActionParams(): ServiceNowActionParams {
+  return {
+    subAction: 'pushToService',
+    subActionParams: {
+      incident: {
+        short_description: MonitorStatusTranslations.defaultActionMessage,
+        description: MonitorStatusTranslations.defaultActionMessage,
+        impact: '2',
```

Comment:
Should we plan to make these fields configurable for this, Jira, PagerDuty, etc?

Reply:
yeah make sense, let's follow up on this.